lua/ttt2/libraries/game_effects.lua

```diff
@@ -0,0 +1,124 @@
+-- @param number spread_force The force that each fire will be flung with.
+-- @param boolean immobile If true, fires will become stationary once they begin burning.
+-- @param number size The physical scale of the fires.
+-- @param number lifetime_variance The amount each lifetime for each fire can vary.
+-- @return table A table full of the fire entities.
+-- @realm shared
+function gameEffects.StartFires(pos, tr, num, lifetime, explode, dmgowner, spread_force, immobile, size, lifetime_variance)
+	local flames = {}
+	for i = 1, num do
+		local ang = Angle(-math.Rand(0, 180), math.Rand(0, 360), math.Rand(0, 360))
+		local vstart = pos + tr.HitNormal * 64
+		local ttl = lifetime + math.Rand(-lifetime_variance, lifetime_variance)
+
+		if CreateVFireBall then
+			flames[#flames + 1] = CreateVFireBall(ttl, size, vstart, ang:Forward() * spread_force, dmgowner)
+			continue
+		end
+
+		local flame = ents.Create("ttt_flame")
+		flame:SetPos(vstart)
+		flame:SetFlameSize(size)
+		flame:SetLifeSpan(ttl)
+		flame:SetImmobile(immobile)
+
+		if IsValid(dmgowner) and dmgowner:IsPlayer() then
+			flame:SetDamageParent(dmgowner)
+			flame:SetOwner(dmgowner)
+		end
+
+		flame:SetDieTime(CurTime() + ttl)
+		flame:SetExplodeOnDeath(explode)
+		flame:Spawn()
+		flame:PhysWake()
+
+		local phys = flame:GetPhysicsObject()
+
+		if IsValid(phys) then
+			-- the balance between mass and force is subtle, be careful adjusting
+			phys:SetMass(2)
+			phys:ApplyForceCenter(ang:Forward() * spread_force)
+			phys:AddAngleVelocity(Vector(ang.p, ang.r, ang.y))
+		end
+
+		flames[#flames + 1] = flame
+	end
+
+	return flames
+end
+
+---
+-- Creates a single point of fire, with optional vfire support.
+-- @param Vector pos The position to create the fire at.
+-- @param number scale Controls the height of the flame more than its radius. Informs the size.
+-- @param number ignite_delay The time before the fire becomes a full flame.
+-- @param number life_span How long a fire will burn for.
+-- @param nil|Entity owner The creator of the fire.
+-- @param nil|Entity parent The thing to attach the fire to.
+-- @return nil|Entity The fire it created, or nil if it was merged / couldn't be created.
+-- @realm server
+function gameEffects.SpawnFire(pos, scale, ignite_delay, life_span, owner, parent)
+	if CreateVFire then
+		return CreateVFire(nil, pos, vector_up, life_span, owner)
+	end
+
+	local fire = ents.Create("env_fire")
+
+	if not IsValid(fire) then return end
+
+	fire:SetParent(parent)
+	fire:SetOwner(owner)
+	fire:SetPos(pos)
+	--no glow + delete when out + start on + last forever
+	fire:SetKeyValue("spawnflags", tostring(128 + 32 + 4 + 2 + 1))
+	fire:SetKeyValue("firesize", tostring(scale)) -- hardly controls size, hitbox is goofy, impossible to work with
+	fire:SetKeyValue("fireattack", tostring(ignite_delay))
+	fire:SetKeyValue("health", tostring(life_span))
+	fire:SetKeyValue("ignitionpoint", "64")
+	fire:SetKeyValue("damagescale", "0")
```

Comment:
Previously this was negative with a comment that only negative values lead to preventing damage, is this not true anymore?

Reply:
Similar to the `fireattack` parameter, the negative damagescale value still caused the "fake damage" screen pulse -- and having it at 0 doesn't prevent it either. Having it at a flat 0 fits the intent since we're managing the hurtbox ourselves.

Reply:
Then please add a small comment, why this is set to 0.

Reply:
ok